[.NET] Add UUID and optional constructor parameter docs for 6.2 (#513)
neo4j/driver-documentation: dotnet-manual/modules/ROOT/pages/object-mapping.adoc

<.> Because the constructor sets the `Name` and `Age` attributes, the attributes can have private setters.

[#optional-constructor-parameters]
=== Optional constructor parameters

> If a constructor parameter has a C# default value, the mapper honours that default when the value is absent from the record.

Reviewer: ```suggestion
If a constructor parameter has a default value, the mapper uses it when the value is absent from the record.
```

> If the value _is_ present but its value is incompatible with the parameter type, an exception is still thrown.

Reviewer: ```suggestion
If the value _is_ present but its value is incompatible with the parameter type, an exception is thrown.
```

[NOTE]
====

> If `[MappingDefaultValue]` is applied to a parameter that also has a C# default value and the value is missing, the attribute's value takes precedence over the C# default.

Reviewer: ```suggestion
If `[MappingDefaultValue]` is applied to a parameter that also has a default value and the value is missing, the attribute's value takes precedence over the parameter default.
```

If both are present, the attribute value wins. Using both is not recommended as it makes the intended default unclear.
====

.A class with an optional constructor parameter
[source, csharp]
----
var people = await driver.ExecutableQuery(@"
    MERGE (p:Person {name: $name})
    RETURN p.name AS name  // <.>
    ")
    .WithParameters(new { name = "Alice" })
    .WithConfig(new QueryConfig(database: "<database-name>"))
    .ExecuteAsync()
    .AsObjectsAsync<Person>();

Console.WriteLine(people[0].Age);  // 18  // <.>

class Person {
    [MappingConstructor]
    public Person(string name, int age = 18) {
        Name = name;
        Age = age;
    }

    public string Name { get; }
    public int Age { get; }
}
----

<.> The query does not return `age`.

> <.> `age` is absent from the record, so the mapper uses the C# default of `18`.

Reviewer: ```suggestion
<.> `age` is absent from the record, so the constructor's default is used.
```
